### PR TITLE
fix a bug

### PR DESCRIPTION
**Pull request by cosmicshuai**
Fixes #
#1234 
## Proposed Changes

  -  modify the matching rule in syntax json file to fix a crash bug when editing a lg file.
  -  the reason cause the crash is the previous regex matching rule will produce a recursive matching the function name and function parameters. Since the function name could also be a part of function parameter. Current fix simplify the rule matching the parameter by looking for a beginning '(' and ending ')'.

### CODE REVIEW

**Comment by boydc2014** [comment]
Why the previous rule cause crash?

File: packages/LGvscodeExt/syntaxes/lg.tmLanguage.json [hunk 1]
```diff
@@ -80,7 +80,8 @@
 			]
 		},
 		"params": {
-			"match": "\\((?:[^)(]+|\\((?:[^)(]+|\\([^)(]*\\))*\\))*\\)",
+			"begin": "\\(",
```
Comment (boydc2014) [inline packages/LGvscodeExt/syntaxes/lg.tmLanguage.json]: So, we don't use `match` any more, is this begin end can cross line or not?

Comment (cosmicshuai) [reply packages/LGvscodeExt/syntaxes/lg.tmLanguage.json]: yes. it can cross lines.

Comment (boydc2014) [reply packages/LGvscodeExt/syntaxes/lg.tmLanguage.json]: we don't want cross lines, right?

Comment (boydc2014) [reply packages/LGvscodeExt/syntaxes/lg.tmLanguage.json]: Let's leave this to next PR maybe

**Pull request merged**